dialects/hsql/src/main/kotlin/app/cash/sqldelight/dialects/hsql/grammar/hsql.bnf

```diff
@@ -62,6 +62,7 @@ column_constraint ::= [ CONSTRAINT {identifier} ] (
   override = true
 }
 bind_parameter ::= ( 'DEFAULT' | '?' | ':' {identifier} ) {
```

Comment:
i think this is actually the issue - we should instead have two overrides, one for bind parameter:

```
bind_parameter ::= ( '?' | ':' {identifier} ) {
```

and one for the bind experession:

```
bind_expr ::= 'DEFAULT' | {bind_parameter}
```

and then the compiler should only look at `SqlBindParameter` instead of `SqlBindExpr` whenever its doing replacements, so we dont need a mixin or supertype

Reply:
Hm, but how do you get the value `DEFAULT` when it is overwritten in the dialect, but without a mixin or supertypes defined in dialect-api and so available in the compiler? 
Only by using the untyped text, `bindParameter ?: text`?

And at the moment, the compiler replaces `:foo` with `?`, everywhere, so this logic still needs to be somewhere. 

I do want to remove the hardcoded `?` and move it to the dialects, because some dialects does not support `?` at all, eg postgres, the jdbc api replaces `?` with the `$1` etc under the hood. But using postgres without jdbc eg on native, this replacement needs to be done again, first replacing `:foo` to `?` by sqldelight, than replacing `? ` with `$1`. Instead I want to replace `:foo` directly with `$1` in my custom dialect

Reply:
ohhhhhhhhhhhhhhhhhhhh interesting. I didn't know postgres doesn't support `?`, that does change things.

Reply:
BTW I tested this and I don't know why it does not work...
```bnf
overrides ::= bind_expr | bind_parameter

bind_expr ::= DEFAULT | bind_parameter {
  extends = "com.alecstrong.sql.psi.core.psi.impl.SqlBindExprImpl"
  implements = "com.alecstrong.sql.psi.core.psi.SqlBindExpr"
  override = true
}

bind_parameter ::= ( '?' | ':' {identifier} ) {
  extends = "com.alecstrong.sql.psi.core.psi.impl.SqlBindParameterImpl"
  implements = "com.alecstrong.sql.psi.core.psi.SqlBindParameter"
  override = true
}
```
`PsiErrorElement:<expr> expected, got 'DEFAULT'`, but `bindExpr` is part of `expr` and `DEFAULT` is part of `bindExpr` 🤔

Reply:
Overwriting BindExpr also does not work, because SqlBindExpr is defined in sql-psi and its 'bindParameter` is NotNull. When overwriting the bindExpr in the dialect it would still inherit from SqlBindExpr and would make bindParameter now nullable, which is not supported and results into false positive (bindParamter?.).